Mathlib/Order/GameAdd.lean

```diff
@@ -19,36 +19,68 @@ This file defines, given relations `rα : α → α → Prop` and `rβ : β →
 subsequency relation on the addition of combinatorial games.
 
+We also define `Sym2.GameAdd`, which is the unordered pair analog of `Prod.GameAdd`.
+
 ## Main definitions and results
 
 - `Prod.GameAdd`: the game addition relation on ordered pairs.
 - `WellFounded.prod_gameAdd`: formalizes induction on ordered pairs, where exactly one entry
   decreases at a time.
 
-## Todo
-
-- Add custom `induction` and `fix` lemmas.
-- Define `Sym2.GameAdd`.
+- `Sym2.GameAdd`: the game addition relation on unordered pairs.
+- `WellFounded.sym2_gameAdd`: formalizes induction on unordered pairs, where exactly one entry
+decreases at a time.
 -/
 
 
-variable (rα : α → α → Prop) (rβ : β → β → Prop)
+variable {α β : Type _} {rα : α → α → Prop} {rβ : β → β → Prop}
 
 namespace Prod
 
+variable (rα rβ)
+
 /-- `Prod.GameAdd rα rβ x y` means that `x` can be reached from `y` by decreasing either entry with
   respect to the relations `rα` and `rβ`.
 
   It is so called, as it models game addition within combinatorial game theory. If `rα a₁ a₂` means
   that `a₂ ⟶ a₁` is a valid move in game `α`, and `rβ b₁ b₂` means that `b₂ ⟶ b₁` is a valid move
   in game `β`, then `GameAdd rα rβ` specifies the valid moves in the juxtaposition of `α` and `β`:
   the player is free to choose one of the games and make a move in it, while leaving the other game
-  unchanged. -/
+  unchanged.
+
+  See `sym2.game_add` for the unordered pair analog. -/
+
 inductive GameAdd : α × β → α × β → Prop
   | fst {a₁ a₂ b} : rα a₁ a₂ → GameAdd (a₁, b) (a₂, b)
   | snd {a b₁ b₂} : rβ b₁ b₂ → GameAdd (a, b₁) (a, b₂)
 #align prod.game_add Prod.GameAdd
 
+theorem gameAdd_iff {rα rβ} {x y : α × β} :
+    GameAdd rα rβ x y ↔ rα x.1 y.1 ∧ x.2 = y.2 ∨ rβ x.2 y.2 ∧ x.1 = y.1 :=
+  by
+  constructor
+  · rintro (@⟨a₁, a₂, b, h⟩ | @⟨a, b₁, b₂, h⟩)
+    exacts[Or.inl ⟨h, rfl⟩, Or.inr ⟨h, rfl⟩]
+  · revert x y
+    rintro ⟨a₁, b₁⟩ ⟨a₂, b₂⟩ (⟨h, rfl : b₁ = b₂⟩ | ⟨h, rfl : a₁ = a₂⟩)
+    exacts[GameAdd.fst h, GameAdd.snd h]
+#align prod.game_add_iff Prod.gameAdd_iff
+
+theorem gameAdd_mk_iff {rα rβ} {a₁ a₂ : α} {b₁ b₂ : β} :
+    GameAdd rα rβ (a₁, b₁) (a₂, b₂) ↔ rα a₁ a₂ ∧ b₁ = b₂ ∨ rβ b₁ b₂ ∧ a₁ = a₂ :=
+  gameAdd_iff
+#align prod.game_add_mk_iff Prod.gameAdd_mk_iff
+
+@[simp]
+theorem gameAdd_swap_swap : ∀ a b : α × β, GameAdd rβ rα a.swap b.swap ↔ GameAdd rα rβ a b :=
+  fun ⟨a₁, b₁⟩ ⟨a₂, b₂⟩ => by rw [Prod.swap, Prod.swap, gameAdd_mk_iff, gameAdd_mk_iff, or_comm]
+#align prod.game_add_swap_swap Prod.gameAdd_swap_swap
+
+theorem gameAdd_swap_swap_mk (a₁ a₂ : α) (b₁ b₂ : β) :
+    GameAdd rα rβ (a₁, b₁) (a₂, b₂) ↔ GameAdd rβ rα (b₁, a₁) (b₂, a₂) :=
+  gameAdd_swap_swap rβ rα (b₁, a₁) (b₂, a₂)
+#align prod.game_add_swap_swap_mk Prod.gameAdd_swap_swap_mk
+
 /-- `GameAdd` is a subrelation of `Prod.Lex`. -/
 theorem gameAdd_le_lex : GameAdd rα rβ ≤ Prod.Lex rα rβ := fun _ _ h =>
   h.rec (Prod.Lex.left _ _) (Prod.Lex.right _)
```

Comment:
The docstring on line 89 was updated in Lean 3 too

Reply:
:+1: